configure.ac

```diff
@@ -82,20 +82,20 @@ case $system in
     ;;
 esac
 
-CCFLAGS="$CCFLAGS $OPENMP_CFLAGS"
-
 if test $ENABLE_DEBUG = yes; then
-  CCFLAGS="$CCFLAGS -g"
+  DEBUG_CFLAGS="-g"
 fi
 
 if test $ENABLE_PIC = yes; then
-  CCFLAGS="$CCFLAGS -fPIC"
+  PIC_CFLAGS="-fPIC"
 fi
 
 if test $ENABLE_MPI = yes; then
-  CCFLAGS="$CCFLAGS -DUSE_MPI"
+  MPI_CFLAGS="-DUSE_MPI"
 fi
 
+CCFLAGS="$CCFLAGS $DEBUG_CFLAGS $OPENMP_CFLAGS $PIC_CFLAGS $MPI_CFLAGS"
```

Comment:
I'll remove my erronous comments from GitHub for clarity, I guess you'll have gotten them in email.